### PR TITLE
Initial support adding a new pass with dummy output

### PR DESCRIPTION
**Pull request by tcwanggoogle**

### CODE REVIEW

File: include/llvm/Transforms/IPO/StructFieldCacheAnalysis.h [hunk 1]
```diff
@@ -0,0 +1,30 @@
+//===-- StructFieldCacheAnalysis.h - Performs Cache Aware Structure Analysis- C++ -*-===//
+//
+//                     The LLVM Compiler Infrastructure
+//
+// This file is distributed under the University of Illinois Open Source
+// License. See LICENSE.TXT for details.
+//
+//===----------------------------------------------------------------------===//
+//
+// This pass exports all llvm.bitset's found in the module in the form of a
```
Comment (cmtice) [inline include/llvm/Transforms/IPO/StructFieldCacheAnalysis.h]: I don't believe this comment is correct.  Did you copy/paste it?

Comment (cmtice) [inline include/llvm/Transforms/IPO/StructFieldCacheAnalysis.h:10]: I'm not sure this comment is correct?  Might this be a copy/paste error?

File: lib/Transforms/IPO/StructFieldCacheAnalysis.cpp [hunk 2]
```diff
@@ -0,0 +1,77 @@
+#include "llvm/Transforms/IPO/StructFieldCacheAnalysis.h"
```
Comment (cmtice) [inline lib/Transforms/IPO/StructFieldCacheAnalysis.cpp:1]: This file is missing the basic license info & descriptive comment at the top.

File: include/llvm/Transforms/IPO/StructFieldCacheAnalysis.h (hunk 1 above)
Comment (tcwanggoogle) [reply include/llvm/Transforms/IPO/StructFieldCacheAnalysis.h:10]: Oh, I am sorry that I forgot to change the comment in the header file. I copied it from another pass.

File: lib/Transforms/IPO/StructFieldCacheAnalysis.cpp (hunk 2 above)
Comment (tcwanggoogle) [reply lib/Transforms/IPO/StructFieldCacheAnalysis.cpp:1]: I'll add them. Thanks!

**Pull request closed without merging**